client/windows/QuickRecruitmentWindow.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * QuickRecruitmentWindow.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+#include "QuickRecruitmentWindow.h"
+#include "../../lib/mapObjects/CGTownInstance.h"
+#include "../CPlayerInterface.h"
+#include "../widgets/Buttons.h"
+#include "../gui/CGuiHandler.h"
+#include "../../CCallback.h"
+#include "../CreatureCostBox.h"
+#include "../lib/ResourceSet.h"
+#include "CreaturePurhaseCard.h"
+
+
+void QuickRecruitmentWindow::setButtons()
+{
+	setCancelButton();
+	setBuyButton();
+	setMaxButton();
+}
+
+void QuickRecruitmentWindow::setCancelButton()
+{
+	cancelButton = std::make_shared<CButton>(Point((pos.w / 2) + 47, 417), "ICN6432.DEF", CButton::tooltip(), [&](){ close(); }, SDLK_RETURN);
+	cancelButton->setImageOrder(0, 1, 2, 3);
+}
+
+void QuickRecruitmentWindow::setBuyButton()
+{
+	buyButton = std::make_shared<CButton>(Point((pos.w/2)-33, 417), "IBY6432.DEF", CButton::tooltip(), [&](){ purhaseUnits(); });
+	cancelButton->assignedKeys.insert(SDLK_ESCAPE);
+	buyButton->setImageOrder(0, 1, 2, 3);
+}
+
+void QuickRecruitmentWindow::setMaxButton()
+{
+	maxButton = std::make_shared<CButton>(Point((pos.w/2)-113, 417), "IRCBTNS.DEF", CButton::tooltip(), [&](){ maxAllCards(cards); });
+	maxButton->setImageOrder(0, 1, 2, 3);
+}
+
+void QuickRecruitmentWindow::setCreaturePurhaseCards()
+{
+	Point pos = Point(8,64);
```

Comment:
With only one creature available this looks ugly. Card is aligned to left, and too much empty space after it.